File: app/src/main/java/com/eurogame/scorecard/domain/model/Game.kt
package com.eurogame.scorecard.domain.model

data class Game(
    val id: Long = 0,
    val name: String,
    val description: String,
    val players: List<Player> = emptyList(),
    val categories: List<ScoreCategory> = emptyList(),
    val scores: Map<Pair<Long, Long>, Int> = emptyMap(), // (playerId, categoryId) -> score
    val createdAt: Long = System.currentTimeMillis(),
    val isActive: Boolean = true
)

File: src/types/game.ts
export interface ScoreCategory {
  id: string;
  title: string;
  subtitle?: string;
}

export interface Player {
  id: string;
  name: string;
  scores: Record<string, number>; // categoryId -> score
}

export interface GameSetup {
  gameName: string;
  description: string;
  players: Player[];
  categories: ScoreCategory[];
}

export interface GameState extends GameSetup {
  isGameStarted: boolean;
}
